### PR TITLE
#162192779 Throws an error if environment variables are not configured

### PR DESCRIPTION
**Pull request by Collins33**
#### What does this PR do?
Throws an error if the environment variables are not configured

#### Description of Task to be completed?
Add check on the modules that throws an error if one of the required variables is not there

#### How should this be manually tested?
- Checkout to this branch.
- Remove the slack token from the env file
- Run the app using `npm start`.
- An error should be thrown saying `Some variables are missing`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#162192779](https://www.pivotaltracker.com/story/show/162192779)

#### Screenshots (If applicable)
N/A

### CODE REVIEW

File: server/modules/freckle/projects.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ dotenv.config();
 const freckleUrl = 'https://api.letsfreckle.com/v2';
 const freckleToken = process.env.FRECKLE_ADMIN_TOKEN;
 
+
```
Comment (IAMOTZ) [inline server/modules/freckle/projects.js]: Please remove this extra white space.

File: server/modules/slack/slackIntegration.js [hunk 2]
```diff
@@ -5,7 +5,9 @@ import makeChannelNames from '../../helpers/slackHelpers';
 
 dotenv.config();
 
+
```
Comment (IAMOTZ) [inline server/modules/slack/slackIntegration.js]: Ditto ...

File: server/modules/slack/slackIntegration.js [hunk 3]
```diff
@@ -5,7 +5,9 @@ import makeChannelNames from '../../helpers/slackHelpers';
...
 const { SLACK_TOKEN } = process.env;
+
```
Comment (IAMOTZ) [inline server/modules/slack/slackIntegration.js]: Ditto ...

File: server/validator.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+const validator = () => {
```
Comment (IAMOTZ) [inline server/validator.js]: This function name is too generic, can we make it more specific???.
From the implementation of the function, what the function seems to be doing is to verify that all the required environment variables are available, so a name like `validateEnvironmentVars` might be better(just a suggestion).

File: server/validator.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    'SLACK_AVAILABLE_DEVS_CHANNEL_ID',
+  ];
+  const missingVariable = [];
+  requiredEnvVariables.forEach((variable) => {
+    if (!process.env[variable]) {
+      missingVariable.push(variable);
+    }
+  });
+  if (missingVariable.length >= 1) {
+    throw Error('Some variables are missing');
```
Comment (IAMOTZ) [inline server/validator.js]: How about you inform the user about the environment variables that are missing???
Something like:
```
throw Error(`${missingVariable.length} environment variables are missing: ${missingVariable.join(', ')}`);
```

File: server/modules/freckle/projects.js (hunk 1 above)
Comment (Collins33) [reply server/modules/freckle/projects.js]: I have removed it

File: server/modules/slack/slackIntegration.js (hunk 2 above)
Comment (Collins33) [reply server/modules/slack/slackIntegration.js]: Removed

File: server/modules/slack/slackIntegration.js (hunk 3 above)
Comment (Collins33) [reply server/modules/slack/slackIntegration.js]: Removed

File: server/validator.js (hunk 4 above)
Comment (Collins33) [reply server/validator.js]: The function has been renamed

File: server/validator.js (hunk 5 above)
Comment (Collins33) [reply server/validator.js]: The response to the user has been updated

**Pull request merged**